src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FTCategoryUpdateServiceImplTest.java

```diff
@@ -0,0 +1,80 @@
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+import java.time.Instant;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+public class FTCategoryUpdateServiceImplTest {
+
+    private static final long VALID_ID = 1L;
+
+    private static final long INVALID_ID = 10L;
+
+    private static final String NAME = "test_name";
+
+    private static final FinancialTransactionType TYPE_INCOME = FinancialTransactionType.INCOME;
+
+    private static final FinancialTransactionCategory VALID_CATEGORY
+            = FinancialTransactionCategory.builder()
+            .id(VALID_ID)
+            .name(NAME)
+            .type(TYPE_INCOME)
+            .creationDate(Instant.now())
+            .build();
+
+    private static final FinancialTransactionCategoryDTO VALID_CATEGORY_DTO
+            = new FinancialTransactionCategoryDTO(VALID_ID, NAME, TYPE_INCOME);
+
+    private static final FinancialTransactionCategoryUpdateDTO VALID_UPDATE_CATEGORY_DTO
+            = new FinancialTransactionCategoryUpdateDTO(NAME, TYPE_INCOME);
+
+    @Mock
+    private FinancialTransactionCategoryRepository repository;
+
+    @Mock
+    private FinancialTransactionCategoryModelMapper mapper;
+
+    @InjectMocks
+    private FinancialTransactionCategoryServiceImpl service;
+
+    @Test
+    @DisplayName("update the FT category and return the category object if the ID is correct")
+    void testUpdateFTCategoryById_WhenIdIsCorrect_ThenReturnCategoryEntity() {
+        Mockito.when(repository.findById(VALID_ID)).thenReturn(Optional.of(VALID_CATEGORY));
+        Mockito.when(mapper.mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(ArgumentMatchers.any())).thenReturn(VALID_CATEGORY_DTO);
+        FinancialTransactionCategoryDTO dto
+                = service.updateFinancialTransactionCategory(VALID_ID, VALID_UPDATE_CATEGORY_DTO);
+        Assertions.assertEquals(dto, VALID_CATEGORY_DTO);
+    }
+
+    @Test
+    @DisplayName("do not update FT Category if id is incorrect then throw AppRuntimeException and does not update entity")
+    void testUpdateFTCategoryById_WhenIdIsIncorrect_ThenThrowError() {
+        Mockito.when(repository.findById(INVALID_ID)).thenReturn(Optional.empty());
+        AppRuntimeException exception = Assertions.assertThrows(
+                AppRuntimeException.class,
+                () -> service.updateFinancialTransactionCategory(INVALID_ID, VALID_UPDATE_CATEGORY_DTO)
+        );
+        Assertions.assertEquals(ErrorCode.FTC001.getBusinessStatusCode(), exception.getBusinessStatusCode());
+        Mockito.verify(repository, Mockito.never()).save(ArgumentMatchers.any());
```

Comment:
I think it would be great to use static imports for method like never() and so on